vng/vector/primitive.go

```diff
@@ -31,7 +32,8 @@ type PrimitiveWriter struct {
 
 func NewPrimitiveWriter(typ zed.Type, spiller *Spiller, useDict bool) *PrimitiveWriter {
 	var dict map[string]uint32
-	if useDict {
+	// TODO https://github.com/brimdata/zed/issues/4839
+	if useDict && os.Getenv("ZED_USE_DICT") != "" {
```

Comment:
Can we make this `os.Getenv("ZED_NO_DICT") == ""` so we don't have to change anything in vng/ztests/?

Reply:
We can, but I was trying to avoid problems like when you hit a broken file during perf testing. I think it makes sense to default to not producing broken files?

Reply:
Ideally this is temporary and we remove the flag when dict encoding is fixed.

Reply:
Oh, also go test automatically runs fuzzer testdata, so we might have to put `ZED_NO_DICT=1 go test ./...` in the makefile to avoid fuzzer crashes.

Reply:
Let's merge #4682 and then drop ZED_USE_DICT.

Reply:
I assume you mean https://github.com/brimdata/zed/pull/4862. That sounds good. I'll try fuzzing on that branch to make sure it's sufficient.

Reply:
Beware: When I tried fuzzing with #4862 I immediately trip over the problem fixed in #4861.